keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,682 @@
+
+  - Create/Delete a namespace
+  - Create a ReplicaSet in the new namespace
+
+
+  Test setup
+
+  - Create a new namespace
+
+
+  Test steps
+
+  - Create a ReplicaSet, with volumes of types of
+    `configMap`, `secret`, `downwardAPI`, `emptyDir`, `persistentVolumeClaim`, `projected`
+  - Poll the ReplicaSet and the namespace, and a new Pod should be created
+  - Create a ReplicaSet, with a volume with type none of
+    `configMap`, `secret`, `downwardAPI`, `emptyDir`, `persistentVolumeClaim`, `projected`
+  - Poll the ReplicaSet, should got `ReplicaFailure` condition from `status`, and no Pod is created
+
+
+  Test teardown
+
+  - Delete the namespace
+
+#### Run as root
+
+  *Conformance Profile: Isolated*
+
+  Containers in Pod created without explicit RBAC associated with the Pod service account,
+  should run as non-root user.
+
+  _Note: this may change if a better isolation runtime (e.g. VM) is used for the container._
+
+###### Test Recommendation
+
+  This test requires a container image with the programs (e.g. `id`) to get current UID and GID.
+
+  The test code consumes Kubernetes API as a serviceaccount/user with privileges:
+
+  - Create/Delete a namespace
+  - Create a ReplicaSet in the new namespace
+
+
+  Test setup
+
+  - Create a new namespace
+
+
+  Test steps
+
+  - Create a ReplicaSet
+  - Poll the ReplicaSet and the namespace, and a new Pod should be created
+  - Check the output of the Pod (can be log) and make sure UID and GID are not zero
+
+
+  Test teardown
+
+  - Delete the namespace
+
+#### Network Isolation
+
+  *Conformance Profile: Isolated*
+
+  Ingress/Egress are blocked for all cross-namespace traffic for all Pods by default,
+  except the Egress to access `kube-dns`.
+  Namespaces `kube-system` and those not selected are not touched.
+
+###### Test Recommendation
+
+  This test requires a container image with the programs (e.g. `socat`) to communicate over network.
+  Each container will have a listening port open to receive messages and also a process periodically sending messages to an remote address.
+  The received messages will be printed on stdout.
+  The container image will auto perform a test to connect to kubelet read-only port.
+
+  The test code consumes Kubernetes API as a serviceaccount/user with privileges:
+
+  - Create/Delete a namespace
+  - Create Pods in the new namespace
+
+
+  Test setup
+
+  - Create two namespaces, ns1 and ns2
+
+
+  Test steps
+
+  - Create a Pod in ns1 to connect to kubelet read-only port
+  - Check the output to make sure the connection should fail;
+  - Create two Pods in ns1
+  - Check the output to make sure they can communicate with each other
+  - Create two Pods, one in ns1, and the other in ns2
+  - Check the output to make sure they can't talk to each other.
+
+
+  Test teardown
+
+  - Delete the namespaces
+
+#### API server configuration
+
+  *Conformance Profile: Essential*
+
+  The following are required:
+
+  - All communication over mTLS/TLS
```

Comment:
nit: just TLS

Reply:
I mean mTLS or TLS. It's OK to setup with stronger security both sides authenticate each other's certificates.